src/main/java/core/basesyntax/Storage.java

```diff
@@ -0,0 +1,25 @@
+package core.basesyntax;
+
+import java.util.HashMap;
+import java.util.Map;
+
+public class Storage {
+
+    private Map<String, Integer> fruits = new HashMap<>();
+
+    public void setFruitQuantity(String fruit, int quantity) {
+        fruits.put(fruit, quantity);
+    }
+
+    public void addFruitQuantity(String fruit, int quantity) {
+        fruits.put(fruit, fruits.getOrDefault(fruit, 0) + quantity);
+    }
+
+    public void reduceFruitQuantity(String fruit, int quantity) {
+        fruits.put(fruit, fruits.getOrDefault(fruit, 0) - quantity);
+    }
+
+    public Map<String, Integer> getAllFruits() {
+        return fruits;
+    }
```

Comment:
To follow the encapsulation principle, you should return a copy of your map or an unmodifiable view of it, to prevent unintended modifications of the Map outside the class. You can use `Collections.unmodifiableMap(fruits)` to return an unmodifiable view of the Map.